projects/packages/connection/src/class-rest-connector.php

```diff
@@ -261,6 +261,29 @@ public function __construct( Manager $connection ) {
 			)
 		);
 
+		// Provider-specific authorization URL endpoint
+		register_rest_route(
+			'jetpack/v4',
+			'/connection/authorize_url/(?P<provider>[a-zA-Z]+)',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'connection_authorize_url_provider' ),
+				'permission_callback' => '__return_true', // Just for testing, it should be user_connection_data_permission_check
+				'args'                => array(
```

Comment:
There will be one more provider, implemented here: https://github.com/Automattic/wp-calypso/pull/101601

It will be under the endpoint: `/log-in/jetpack/link`
And will require two query params: `email_address=<sanitized_url>&auto_trigger`

Reply:
Thanks! The `email_address` comes from an e-mail input, right? Or it should be e-mail address of the WP logged in user.

Reply:
It does come from email input!